zerver/lib/streams.py

```diff
@@ -146,6 +146,15 @@ def create_streams_if_needed(
     added_streams: List[Stream] = []
     existing_streams: List[Stream] = []
     for stream_dict in stream_dicts:
+        if acting_user is not None:
```

Comment:
Triple nesting of `if` statements not required. Since there are no `else` branches to any of these `ifs` you can do away with just a single `if` statement talking about all the conditions and remove triple nesting.